QModManager/API/Messages.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace QModManager.API
+{
+    public abstract class MessageReceiver
+    {
+        public IQMod From;
+
+        public MessageReceiver(IQMod from) => From = from ?? throw new ArgumentNullException("from", "The provided mod is null!");
+        public MessageReceiver(Assembly fromAssembly) => From = QModAPI.GetMod(fromAssembly ?? throw new ArgumentNullException(nameof(fromAssembly), "The provided assembly is null!"), true, true) ?? throw new ArgumentException("The provided assembly is not a mod assembly!", nameof(fromAssembly));
+        public MessageReceiver(string fromID) => From = QModAPI.GetMod(fromID ?? throw new ArgumentNullException(nameof(fromID), "The provided ID is null!"), true, true) ?? throw new ArgumentException("No mod matching the provided ID was found!");
```

Comment:
Should we not throw an exception if the ID is not found?

Reply:
If the ID is not found, it may be because the mod is missing. In that case, I'm not entirely sure an exception should be thrown